### ISSUE TITLE
Invalid handling of AfterControlStatement: MultiLine

### ISSUE THREAD

**Issue by varqox** [llvm/llvm-project#45978]
|  |  |
| --- | --- |
| Bugzilla Link | [46633](https://llvm.org/bz46633) |
| Version | 10.0 |
| OS | Linux |

## Extended Description 

With the following .clang-format file:
```
Language:        Cpp
BreakBeforeBraces: Custom
BraceWrapping:
  AfterControlStatement: MultiLine
ColumnLimit:     80
IndentWidth:     4
```

running clang-format on
```
void foo() {
    if (very_long_variable_name or other_very_long_variable_name or
        long_variable)
    {
        return;
    }
}
```

produces:
```
void foo() {
    if (very_long_variable_name or other_very_long_variable_name or
        long_variable) {
        return;
    }
}
```

Expected output: the same as input.
Command used: clang-format bug.cc
clang-format version: 10.0.0

Additional notes:
If you change "long_variable)" to "long_variable123)" or "long_variab)" formatting is correct -- the brace gets wrapped. So it seems that if an if condition ends slightly after the column limit (if placed in a single line), then it is formatted incorrectly.

**Comment by mydeveloperday** [comment]
The calculation of the TotalLength is off (78) hence why its merging 1 line instead of none... 

During this check the TotalLengh is 78, hence why its not merging on multiline

```
return (Style.ColumnLimit == 0 ||
              TheLine->Last->TotalLength <= Style.ColumnLimit)
                 ? 1
                 : 0;
```

But 78 is here

![image](https://user-images.githubusercontent.com/16046346/146165538-119b700e-89ad-4f2f-b02f-a5d974170e2d.png)

Hence making it longer pushed it over the lint and we don't merge the braces up into this line

if I add another 3 characters it will wrap correctly, i.e. I have to bring the line out to 88 in order to get the TotalLength to be 81

```
void foo() {
    if (very_long_variable_name or other_very_aaalong_variable_name or
        long_variable)
    {
        return;
    }
}

void foo() {
    if (very_long_variable_name || aaaother_very_long_variable_name ||
        long_variable)
    {
        return;
    }
}
```

**Comment by owenca** [comment]
Fixed in f4d52cad.
